source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -497,47 +497,21 @@ void Filter::onPoolFailure(ConnectionPool::PoolFailureReason reason,
   }
 }
 
-void Filter::onPoolReadyBase(Upstream::HostDescriptionConstSharedPtr& host,
-                             const Network::Address::InstanceConstSharedPtr& local_address,
-                             Ssl::ConnectionInfoConstSharedPtr ssl_info) {
-  upstream_handle_.reset();
+void Filter::onPoolReady(const TcpProxy::GenericUpstreamSharedPtr& upstream,
+                         Upstream::HostDescriptionConstSharedPtr& host,
+                         const Network::Address::InstanceConstSharedPtr& local_address,
+                         StreamInfo::StreamInfo& info) {
```

Comment:
this can be const?

Reply:
add 
`FilterStateConstSharedPtr sharedFilterState()`
and
`void setConstUpstreamFilterState(const FilterStateConstSharedPtr& filter_state)` to address this.

The only cast used is from `shared_ptr<FilterState>` to `shared_ptr<const FilterState>`.